src/mappings/referrer.ts

```diff
@@ -17,54 +19,69 @@ export function handleReferralCodeCreated(event: OnReferralCodeCreated): void {
 }
 
 export function handleReferralCodeUpserted(event: OnReferralCodeUpserted): void {
-    let existingReferralCode = getReferralCode(event.params.oldReferralCode)
-    let trader = getOrCreateTrader(event.params.addr)
-    // Only proceed any action, if the referee has a code attached
-    if (existingReferralCode) {
-        // Referee removes code
-        if (event.params.action == 1) {
-            // trader no longer has a referee code
-            trader.refereeCode = null
-            trader.save()
-
-            // update referee count for code
-            existingReferralCode.numReferees = existingReferralCode.numReferees.minus(BI_ONE)
-            existingReferralCode.save()
-            return
-        }
+    if (!event.params.oldReferralCode) {
+        return
+    }
 
-        // add and update should only be possible if newReferralCode contains a value
-        if (event.params.newReferralCode != "") {
-            let newReferralCode = getReferralCode(event.params.newReferralCode)
-            let newReferralCodeDayData = getReferralCodeDayData(event, newReferralCode.id)
-            if (event.params.action == 2) {
-                // nothing changes here
-                if (event.params.oldReferralCode == event.params.newReferralCode) return
-            }
+    const oldRefCode = getReferralCode(event.params.oldReferralCode)
```

Comment:
Suggest to load old / new referral code in functions like `removeReferralCodeFromReferee()` so we can do different check for different action, i.e.,

`remove()`: load old code and check if old code exists.
`replace()`: load old and new code and check if both codes exist.
`create()`: load new code and check if new code exists.

Reply:
refactored